README.md

```diff
@@ -10,17 +10,35 @@ This GitHub Action (created using typescript) uses [CLI for Microsoft 365](https
 Create a workflow `.yml` file in your `.github/workflows` directory. An [example workflow](#example-workflow---cli-for-microsoft-365-login) is available below. For more information, reference the GitHub Help Documentation for [Creating a workflow file](https://help.github.com/en/articles/configuring-a-workflow#creating-a-workflow-file).
 
 ### Inputs
-- `ADMIN_USERNAME` : **Required** Username (email address of the admin)
-- `ADMIN_PASSWORD` : **Required** Password of the admin
+- `ADMIN_USERNAME` : Username (email address of the admin)
+- `PASSWORD` : Password for the admin or the certificate
+- `CERTIFICATE_ENCODED` : Base64-encoded string with certificate private key
+- `AAD_APP_ID` : App ID of the Azure AD application to use for certificate authentication
+- `TENANT_ID` : ID or domain (for example "contoso.onmicrosoft.com") of the tenant from which accounts should be able to authenticate
+
+All inputs are optional. But depending of the authentication type chosen, following pair of inputs will be necessary:
+
+- `authType` is `password`: `ADMIN_USERNAME` and `PASSWORD` are required
+- `authType` is `certificate`: at least `CERTIFICATE_ENCODED` and `AAD_APP_ID` are required
+  - Depending on your tenant environment, `TENANT_ID` will be required
+  - Depending on the certificate provided, if encoded with password, `PASSWORD` will be required
 
 #### Optional requirement
-Since this action requires user name and password which are sensitive pieces of information, it would be ideal to store them securely. We can achieve this in a GitHub repo by using [secrets](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets). So, click on `settings` tab in your repo and add 2 new secrets:
-- `adminUsername` - store the admin user name in this (e.g. user@contoso.onmicrosoft.com)
-- `adminPassword` - store the password of that user in this.
+Since this action requires sensitive information such as user name, password and encoded certificate for example, it would be ideal to store them securely. We can achieve this in a GitHub repo by using [secrets](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets). So, click on `settings` tab in your repo and add:
+
+- 2 new secrets if `authType` is `password`:
+  - `adminUsername` - store the admin user name in this (e.g. user@contoso.onmicrosoft.com)
+  - `adminPassword` - store the password of that user in this.
+
+- 3 new secrets if `authType` is `certificate`:
+  - `AppID` - store the Azure AD application ID used for authentication
```

Comment:
Is there any reason why we suggest naming first 2 secrets starting with lowercase and these 3 secrets with uppercase?

Reply:
No reason at all, just inconstitency 🤷‍♂️

Let me adjust this.

I didn't find any stated naming convention except the Actions REST API [code sample](https://docs.github.com/en/rest/reference/actions#list-organization-secrets--code-samples) which returns full uppercase secrets. So maybe we could follow this?

Reply:
I'd say, let's align with the existing documentation so that we're not causing any breaking changes. We can always take a note of that to adjust in the next major version.